src/ucs/sys/iovec.h

```diff
@@ -59,6 +59,27 @@ void ucs_iov_advance(struct iovec *iov, size_t iov_cnt,
  */
 size_t ucs_iov_get_max();
 
+/**
+ * Calculates the total length of the iov array buffers.
+ *
+ * @param [in]     iov            A pointer to an array of iovec elements.
+ * @param [in]     iov_cnt        A number of elements in a iov array.
+ *
+ * @return The amount, in bytes, of the data that is stored in the iov
+ *         array buffers.
+ */
+static UCS_F_ALWAYS_INLINE
+size_t ucs_iov_total_length(const struct iovec *iov, size_t iov_cnt)
```

Comment:
why not in C file? err handling is slow path

Reply:
`ucs_iov_total_length` is used in fast path if `iov_cnt > 0`, but a user is sending `0`-byte payload

Reply:
we are calling it only if ret==0

Reply:
AFAIK sending 0-byte payload is not fast path
also checking iov_cnt separately is probably redundant..

Reply:
> AFAIK sending 0-byte payload is not fast path
> also checking iov_cnt separately is probably redundant..

I added check for iov_cnt to avoid calculating total length if iov_cnt == 0

> AFAIK sending 0-byte payload is not fast path

why? I think it has to be fast path, since user may want to send 0-byte message and it is handled right after iov_func/io_func is completed

Reply:
in which case we can send 0 byte message in TCP?

Reply:
> in which case we can send 0 byte message in TCP?

there is no such case currently, but we should support this case, right? since this is a common API that can be used later in other components

Reply:
we can support but it doesn't have to be so optimized especially in tcp case

Reply:
pls move this to C file

Reply:
done